qt/widgets/common/src/FitScriptGeneratorModel.cpp

```diff
@@ -157,148 +108,415 @@ void FitScriptGeneratorModel::addWorkspaceDomain(
+    std::string const &fullParameter, std::string const &fullTie) const {
+  if (fullTie.empty() || isNumber(fullTie) || !validTie(fullTie))
+    return fullTie;
+
+  if (m_fittingMode == FittingMode::SEQUENTIAL)
+    return fullTie;
+
+  auto const domainIndex = findDomainIndex(workspaceName, workspaceIndex);
+  return getEquivalentParameterTieForDomain(domainIndex, fullParameter,
+                                            fullTie);
+}
+
+std::string FitScriptGeneratorModel::getEquivalentParameterTieForDomain(
+    FitDomainIndex domainIndex, std::string const &fullParameter,
+    std::string const &fullTie) const {
+  auto const parameterDomainIndex = getFunctionIndexAt(fullParameter, 0);
+  auto const tieDomainIndex = getFunctionIndexAt(fullTie, 0);
+  if (parameterDomainIndex == tieDomainIndex)
+    return replaceTopFunctionIndexWith(fullTie, domainIndex.value);
+  return fullTie;
+}
+
+std::string FitScriptGeneratorModel::getAdjustedFunctionIndex(
+    std::string const &parameter) const {
+  if (parameter.empty() || isNumber(parameter))
+    return parameter;
+
+  if (m_fittingMode == FittingMode::SEQUENTIAL)
+    return parameter;
+  return removeTopFunctionIndex(parameter);
+}
+
+void FitScriptGeneratorModel::updateParameterValue(
+    std::string const &workspaceName, WorkspaceIndex workspaceIndex,
+    std::string const &fullParameter, double newValue) {
+  auto const domainIndex = findDomainIndex(workspaceName, workspaceIndex);
+  if (!hasGlobalTie(fullParameter)) {
+    auto const parameter = getAdjustedFunctionIndex(fullParameter);
+    m_fitDomains[domainIndex.value]->setParameterValue(parameter, newValue);
+
+    updateParameterValuesWithLocalTieTo(domainIndex, parameter, newValue);
+    if (m_fittingMode == FittingMode::SIMULTANEOUS) {
+      updateParameterValuesWithGlobalTieTo(fullParameter, newValue);
+    }
+  }
 }
 
-std::pair<double, double>
-FitScriptGeneratorModel::xLimits(MatrixWorkspace_const_sptr const &workspace,
-                                 WorkspaceIndex workspaceIndex) const {
-  if (workspace) {
-    auto const xData = workspace->x(workspaceIndex.value);
-    return std::pair<double, double>(xData.front(), xData.back());
+void FitScriptGeneratorModel::updateParameterValuesWithLocalTieTo(
+    FitDomainIndex domainIndex, std::string const &parameter, double newValue) {
+  for (auto const &tiedParameterName :
+       m_fitDomains[domainIndex.value]->getParametersTiedTo(parameter)) {
+    m_fitDomains[domainIndex.value]->setParameterValue(tiedParameterName,
+                                                       newValue);
   }
+}
 
-  throw std::invalid_argument("The workspace '" + workspace->getName() +
-                              "' is not a matrix workspace.");
+void FitScriptGeneratorModel::updateParameterValuesWithGlobalTieTo(
+    std::string const &fullParameter, double newValue) {
+  // Deep copy so that global ties can be removed whilst in this for loop
```

Comment:
do you mean updated instead of removed?

Reply:
I think I meant removed. If it finds that a global tie has been invalidated, it will remove it - otherwise it will successfully update the parameter value.

I'll add a larger comment to make this clearer, thanks